pds4indextools/pds4_create_xml_index.py

```diff
@@ -183,6 +183,48 @@ def extract_logical_identifier(tree):
     return logical_identifier.text.strip()
 
 
+def match_dict_keys(data, pattern):
+    """
+    Match dictionary keys against a pattern with support for `**` as a recursive wildcard.
+
+    Parameters:
+        data (dict): The dictionary whose keys are to be matched against the pattern.
+                     Keys are expected to be strings.
+        pattern (str): The pattern to match against dictionary keys. Supports Unix
+                       shell-style wildcards including `**` for recursive matching.
+
+    Returns:
+        list: A list of keys from the input dictionary that match the given pattern.
+    """
+    def match_segment(segment, pattern):
+        return fnmatch.fnmatch(segment, pattern)
+
+    def match_recursive_helper(segments, patterns):
+        if not patterns:
+            return not segments
+
+        pattern = patterns[0]
+        if pattern == '**':
+            if match_recursive_helper(segments, patterns[1:]):
+                return True
+            return bool(segments) and match_recursive_helper(segments[1:], patterns)
+        elif segments:
+            return match_segment(segments[0], pattern) and match_recursive_helper(
```

Comment:
This would format better as something like:
```
            return (match_segment(segments[0], pattern) and
                    match_recursive_helper(segments[1:], patterns[1:])
```